pkg/agent/multicast/mcast_controller.go

```diff
@@ -377,11 +456,23 @@ func (c *Controller) addOrUpdateGroupEvent(e *mcastGroupEvent) {
 			c.updateGroupMemberStatus(obj, e)
 		}
 	}
+	return
+}
+
+func (c *Controller) podDeletionHandler(stopCh <-chan struct{}) {
+	for {
+		select {
+		case e := <-c.podDeletionCh:
+			c.removeLocalInterface(e)
+		case <-stopCh:
+			return
+		}
+	}
 }
 
 func podInterfaceIndexFunc(obj interface{}) ([]string, error) {
 	groupState := obj.(*GroupMemberStatus)
-	podInterfaces := make([]string, 0, len(groupState.localMembers))
+	podInterfaces := make([]string, 0)
```

Comment:
maybe keep the cap here to avoid memory re-allocation when calling append?

Reply:
done.